src/elastio-snap.c

```diff
@@ -4201,14 +4213,18 @@ static int __tracer_setup_snap(struct snap_device *dev, unsigned int minor, stru
 	// alloc_disk function has been disappeared starting from the kernel 5.15
 	dev->sd_gd = alloc_disk(1);
 #else
-	dev->sd_gd = __alloc_disk_node(dev->sd_queue, NUMA_NO_NODE, &sd_bio_compl_lkclass);
+	dev->sd_gd = elastio_snap_alloc_disk_node(dev->sd_queue, NUMA_NO_NODE, &sd_bio_compl_lkclass);
```

Comment:
It seems better to use macro [`blk_alloc_disk`](https://elixir.bootlin.com/linux/v5.19.16/source/include/linux/blkdev.h#L826) which is [widely used](https://elixir.bootlin.com/linux/v5.19.16/C/ident/blk_alloc_disk) in kernel.
or exported function [`__blk_alloc_disk`](https://elixir.bootlin.com/linux/v5.19.16/source/block/genhd.c#L1410).
To avoid search of an internal function by address.

Reply:
I disagree. `blk_alloc_disk` and `__blk_alloc_disk` were introduced only after v5.14. This function also creates a queue underneath, which makes us reconsider (remove) queue logic in the driver, which, in its turn, won't be compatible with older drivers. Technically, we can do this, but I don't think this complexity is worth it.

Reply:
>  `blk_alloc_disk` and `__blk_alloc_disk` were introduced only after v5.14.

This is, actually a counter argument, not an argument. We are fixing argument for 5.19. It means the proposed func exists fir a long period of time. 

Another option is to use more high level func `blk_mq_alloc_disk` which is used by `nbd` and by `loopback` device drivers